spec/controllers/account/utility_providers_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe Account::UtilityProvidersController, type: :controller do
+  render_views
+
+  let!(:user) { create(:user) }
+  let!(:company) { create(:company) }
+  let!(:company2) { create(:company) }
+  let!(:utility_provider) { create(:billing_contract, company: company, user: user) }
```

Comment:
since you add associations to billing contact you can write
`let!(:utility_provider) { create(:billing_contract) }`
`let!(:user) { utility_provider.user }`
`company = utility_provider.company`
if you use user only once for #sign_in, you can write 
`before { sign_in utility_provider.user }`
this for future, for now it OK